test/unittests/state_transition.hpp

```diff
@@ -0,0 +1,68 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include <evmone/evmone.h>
+#include <gtest/gtest.h>
+#include <test/state/host.hpp>
+
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+
+namespace evmone::test
+{
+using namespace evmone::state;
+
+/// Fixture to defining test cases in form similar to JSON State Tests.
+///
+/// It takes the "pre" state and produces "post" state by applying the defined "tx" transaction.
+/// Then expectations declared in "except" are checked in the "post" state.
+class state_transition : public testing::Test
+{
+protected:
+    /// The default sender address of the test transaction.
+    /// TODO: We need a private key of this address so pick some other one later.
+    static constexpr auto Sender = 0x8e4d_address;
+
+    /// The default destination address of the test transaction.
+    static constexpr auto To = 0xc0de_address;
+
+    static inline evmc::VM vm{evmc_create_evmone()};
+
+    struct ExpectedAccount
+    {
+        bool exists = true;
+        std::optional<uint64_t> nonce;
+        std::optional<intx::uint256> balance;
+        std::unordered_map<evmc::bytes32, evmc::bytes32> storage;
+    };
+
+    struct Expectation
+    {
+        evmc_status_code status = EVMC_SUCCESS;
+        std::optional<int64_t> gas_used;
+
+        std::unordered_map<evmc::address, ExpectedAccount> post;
+    };
+
+
+    BlockInfo block{.gas_limit = 1'000'000, .base_fee = 10};
```

Comment:
Is this GCC extension to have these initializers?

Reply:
No, this is part of C++20.